Order Creation: Add Product UI

rachelmcr opened the pull request:
Part of: #5408

## Description

This adds the Add Product UI in the order creation flow. Tapping the "Add product" button on the New Order screen opens a modal Add Product view. For now, the UI is a static product row with temporary, fake data; it will eventually be populated with a list of products on the store.

Future PRs will add real data and interactions, as well as a search bar on the Add Product screen.

## Changes

* Moves `ProductRow` out of the `NewOrder` view, so it can be reused in the `AddProduct` view.
* Adds the option to show/hide the stepper in `ProductRow` (shown on the New Order screen and hidden on the Add Product screen).
* Adds the `AddProduct` view, which for now displays a navigation bar and single product row with static data.
* Adds navigation from `NewOrder` to `AddProduct` when you tap on the "Add product" button.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. (If Simple Payments is also enabled, tap "Create order" in the bottom sheet that appears.)
4. On the New Order screen, tap the "Add product" button.
5. Confirm the Add Product modal appears with a product row, and you can tap the Close button or swipe to dismiss it.

## Screenshots


https://user-images.githubusercontent.com/8658164/144036120-9cb70da7-a696-46ae-aac6-3c60b5627eeb.mp4

iPhone|iPad
-|-
![AddProductUI-iPhone](https://user-images.githubusercontent.com/8658164/144036137-0339bdb8-1761-4777-97c9-de94ae667517.png)|![AddProductUI-iPad](https://user-images.githubusercontent.com/8658164/144036143-a1a06dfc-40f0-4072-b54a-4b3e917eae1a.png)

## Submitter Checklist

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

rachelmcr committed:
Move ProductRow for reuse in Add Product view

rachelmcr committed:
Add basic Add Product screen with ProductRow

rachelmcr committed:
Navigate to Add Product screen

rachelmcr committed:
Localize accessibility label for custom stepper

rachelmcr committed:
Only display product quantity stepper after product is added to order

Ecarrion commented on WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift:
```diff
@@ -81,6 +81,7 @@ struct NewOrder: View {
 ///
 private struct ProductsSection: View {
     let geometry: GeometryProxy
+    @State private var showAddProduct: Bool = false
```
Nit:  I know these are self-explanatory but I think it would be good too keep adding a some short docs to it 🪙 🪙

Ecarrion commented on WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/AddProduct.swift:
```diff
@@ -0,0 +1,41 @@
+import SwiftUI
+
+struct AddProduct: View {
```
Same here, a type description would be nice!

Ecarrion reviewed (approved):
This looks good to me! 🚀 

One minor thing, I normally leave this for accessibility review at the end of the project but since you tackled it in previous PRs I will point out that maybe we should adjust the 
`AddProduct` view so dividers can go edge to edge on landscape mode!

<img width="887" alt="edges" src="https://user-images.githubusercontent.com/562080/144052764-46b04ffb-6916-4ae2-b64e-82de581917b3.png">

rachelmcr committed:
Add documentation for views and properties

rachelmcr commented:
Thanks Ernesto! I added more documentation and added a reminder on the issue to polish the edge-to-edge layout before I finish working on this section. Getting the dividers and padding just right is a little fiddly so I'll come back to it once I have all the behavior right.
